wire-ios-sync-engine/Source/Use cases/GetE2eIdentityCertificatesUseCase.swift

```diff
@@ -0,0 +1,118 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+import WireCoreCrypto
+import ASN1Decoder
+
+// sourcery: AutoMockable
+public protocol GetE2eIdentityCertificatesUseCaseProtocol {
+    func invoke(mlsGroupId: MLSGroupID,
+                clientIds: [MLSClientID]) async throws -> [E2eIdentityCertificate]
+}
+
+final public class GetE2eIdentityCertificatesUseCase: GetE2eIdentityCertificatesUseCaseProtocol {
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+
+    public init(coreCryptoProvider: CoreCryptoProviderProtocol) {
+        self.coreCryptoProvider = coreCryptoProvider
+    }
+
+    public func invoke(mlsGroupId: MLSGroupID,
+                       clientIds: [MLSClientID]) async throws -> [E2eIdentityCertificate] {
+
+        let coreCrypto = try await coreCryptoProvider.coreCrypto(requireMLS: true)
+        let clientIds = clientIds.compactMap({$0.rawValue.data(using: .utf8)})
+        let wireIdentities = try await getWireIdentity(coreCrypto: coreCrypto,
+                                                       conversationId: mlsGroupId.data,
+                                                       clientIDs: clientIds)
+        return wireIdentities.isNonEmpty ?  try wireIdentities.compactMap({try $0.toE2eIdenityCertificate()}) : []
```

Comment:
I'm not sure, but do we need to check if `wireIdentities.isNonEmpty`? The fallback is empty array.
Can we just `return try wireIdentities.compactMap({try $0.toE2eIdenityCertificate()}) `?

Reply:
done